common/cache/lru.go

```diff
@@ -239,15 +249,34 @@ func (c *lru) Release(key interface{}) {
 
 // Size returns the number of entries currently in the lru, useful if cache is not full
 func (c *lru) Size() int {
+	if c.activelyEvict {
+		c.EvictItemsPastTimeToLive()
+	}
+
 	c.mut.Lock()
 	defer c.mut.Unlock()
 
 	return len(c.byKey)
 }
 
+// EvictItemsPastTimeToLive evicts all items in the cache which are expired
+// This is not called automatically, but can be called periodically to evict expired items
+func (c *lru) EvictItemsPastTimeToLive() {
+	c.mut.Lock()
+	defer c.mut.Unlock()
```

Comment:
we can avoid multiple lock/unlocks that happens here and on the caller side after this func returns. you can omit locks here and make this a private func. 
nit: also renaming to `evictExpiredItems` sounds simpler.

Reply:
made evictExpiredItems private, then it makes sense to just move the locking for calls to this method to the callers (also done)